x/ratelimit/keeper/msg_server_test.go

```diff
@@ -65,24 +96,52 @@ func (s *KeeperTestSuite) TestMsgServer_RemoveRateLimit() {
 	denom := removeRateLimitMsg.Denom
 	channelId := removeRateLimitMsg.ChannelId
 
-	// TODO: Uncomment once function is implemented
-	_ = denom
-	_ = channelId
-	// // Attempt to remove a rate limit that does not exist
-	// _, err := s.GetMsgServer().RemoveRateLimit(sdk.WrapSDKContext(s.Ctx), removeRateLimitMsg)
-	// s.Require().Error(err)
+	// Attempt to remove a rate limit that does not exist
+	_, err := s.GetMsgServer().RemoveRateLimit(sdk.WrapSDKContext(s.Ctx), removeRateLimitMsg)
+	s.Require().Error(err)
+
+	// Add a rate limit successfully
+	_, err = s.GetMsgServer().AddRateLimit(sdk.WrapSDKContext(s.Ctx), addRateLimitMsg)
+	s.Require().NoError(err)
+
+	_, found := s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
+	s.Require().True(found)
+
+	// Remove the rate limit
+	_, err = s.GetMsgServer().RemoveRateLimit(sdk.WrapSDKContext(s.Ctx), removeRateLimitMsg)
+	s.Require().NoError(err)
+
+	_, found = s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
+	s.Require().False(found)
+}
+
+func (s *KeeperTestSuite) TestMsgServer_ResetRateLimit() {
+	s.SetupTest()
+	validAddr, _ := apptesting.GenerateTestAddrs()
+	addRateLimitMsg.Creator = validAddr
+	resetRateLimitMsg.Creator = validAddr
+
+	denom := resetRateLimitMsg.Denom
+	channelId := resetRateLimitMsg.ChannelId
+
+	// Attempt to reset a rate limit that does not exist
+	_, err := s.GetMsgServer().ResetRateLimit(sdk.WrapSDKContext(s.Ctx), resetRateLimitMsg)
+	s.Require().Error(err)
 
-	// // Add a rate limit successfully
-	// _, err = s.GetMsgServer().AddRateLimit(sdk.WrapSDKContext(s.Ctx), addRateLimitMsg)
-	// s.Require().NoError(err)
+	// Add a rate limit successfully
+	_, err = s.GetMsgServer().AddRateLimit(sdk.WrapSDKContext(s.Ctx), addRateLimitMsg)
+	s.Require().NoError(err)
 
-	// _, found := s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
-	// s.Require().True(found)
+	_, found := s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
+	s.Require().True(found)
 
-	// // Remove the rate limit
-	// _, err = s.GetMsgServer().RemoveRateLimit(sdk.WrapSDKContext(s.Ctx), removeRateLimitMsg)
-	// s.Require().NoError(err)
+	// Reset the rate limit successfully
+	_, err = s.GetMsgServer().ResetRateLimit(sdk.WrapSDKContext(s.Ctx), resetRateLimitMsg)
+	s.Require().NoError(err)
 
-	// _, found = s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
-	// s.Require().False(found)
+	// Check ratelimit quota is reset correctly
+	resetRateLimit, found := s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
+	s.Require().True(found)
+	s.Require().Equal(resetRateLimit.Flow.Inflow, uint64(0))
+	s.Require().Equal(resetRateLimit.Flow.Outflow, uint64(0))
```

Comment:
it might be good to also mint some of that denom with the bank keeper and then add one last check here that Flow.ChannelValue was update to the amount that was minted